### PR TITLE
EVG-19875: Modify ProjectRef struct to include Parsley filters

### PR DESCRIPTION
**Pull request by minnakt**
EVG-19875

### Description
This PR is just to add Parsley filters to the `ProjectRef` and `APIProjectRef` structs. Future tickets will build on this work (including handling validation and saving Parsley filters via GraphQL mutation and PATCH rest route).

#### Note
I used `expression` rather than `name` for the field because I felt like it might be more correct... The filter doesn't really have a name, it's defined by some regex expression. This change can also be made on the frontend.

### Testing
- Added some local data for Parsley filters in the `sys-perf` project and checked that it was correctly loaded into the DB and viewable via http://localhost:9090/rest/v2/projects/sys-perf

### CODE REVIEW

**Review by sophstad — APPROVED** [review]
Lgtm but would probably make sense to have an app reviewer confirm! =]

File: model/project_ref.go
Comment (ablack12) [inline model/project_ref.go:118]: Preference to move this to the bottom since those are also about UI views

**Review by ablack12 — APPROVED** [review]
LGTM with nit

File: model/project_ref.go
Comment (sophstad) [reply model/project_ref.go:118]: I'll wait until this is merged in and follow suit for #6516 👍

**Pull request merged**
